src/pci_virtio_sock.c

```diff
@@ -1302,10 +1304,12 @@ static void handle_connect_fd(struct pci_vtsock_softc *sc, int accept_fd, uint32
 	sock->peer_addr.cid = cid;
 	sock->peer_addr.port = port;
 	sock->local_addr.cid = VMADDR_CID_HOST;
-	/* Start at 2^16 to be larger than a TCP port
+	/* Start at 2^16 to be larger than a TCP port, add a
+	 * generation counter to reduce port reuse.
 	 * XXX Allocate properly.
          */
-	sock->local_addr.port = 65536U + (uint32_t)(sock - &sc->socks[0]);
+	sock->local_addr.port = ((uint32_t)(sock - &sc->socks[0] + 1) << 16)
```

Comment:
The +1 avoids assigning port 0x00000000. Since it is `array-offset + 1` for the zeroeth-sock it'll start at 0x00010000 and allocate through 0x0001ffff and then wrap back to 0x00010000 again, for the first sock 0x00020000... etc